### PR TITLE
Models group3

### PR DESCRIPTION
**Pull request by ryan-schlientz**

### CODE REVIEW

File: BattleSim/src/main/java/com/revature/models/Roles.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+
+/*CREATE TABLE roles (
+    r_id NUMBER(10) PRIMARY KEY,
+    name VARCHAR2(20));
+ * */
```
Comment (ryan-schlientz) [inline BattleSim/src/main/java/com/revature/models/Roles.java]: Remove commented code

File: BattleSim/src/main/java/com/revature/models/Roles.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+public class Roles {
+
+	@Id
+	@Column(name="r_id", updatable=false)
+	@SequenceGenerator(sequenceName="role_id_maker", name="role_id_maker")
+	@GeneratedValue(generator="role_id_maker", strategy=GenerationType.SEQUENCE)
+	private int r_id;
+	
+	private String name;
+
```
Comment (WilRBonilla) [inline BattleSim/src/main/java/com/revature/models/Roles.java:20]: id-less and no-arg constructors missing

File: BattleSim/src/main/java/com/revature/repositories/RolesRepository.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import javax.management.relation.Role;
+
+public interface RolesRepository {
+	public int addRole(Role r);
+	public List<Role> getAllRoles();
+	public Role getRole(int id);
+//	public Role getRole(name);
+	public void updateRoles(Role change);
+	public void deleteRoles(int id);
```
Comment (nmart510) [inline BattleSim/src/main/java/com/revature/repositories/RolesRepository.java]: Plural unnecessary

File: BattleSim/src/main/java/com/revature/repositories/RolesRepository.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+
+import javax.management.relation.Role;
+
+public interface RolesRepository {
+	public int addRole(Role r);
+	public List<Role> getAllRoles();
+	public Role getRole(int id);
+//	public Role getRole(name);
```
Comment (ryan-schlientz) [inline BattleSim/src/main/java/com/revature/repositories/RolesRepository.java]: Remove commented code

File: BattleSim/src/main/java/com/revature/repositories/RolesRepository.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.revature.repositories;
+
+import java.util.List;
+
+import javax.management.relation.Role;
```
Comment (plboronski) [inline BattleSim/src/main/java/com/revature/repositories/RolesRepository.java]: This is the wrong class, updating currently

File: BattleSim/src/main/java/com/revature/repositories/RolesRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
+package com.revature.repositories;
+
+import java.util.List;
+
+import javax.management.relation.Role;
```
Comment (plboronski) [inline BattleSim/src/main/java/com/revature/repositories/RolesRepositoryImpl.java]: Wrong import

File: BattleSim/src/main/java/com/revature/models/Roles.java (hunk 2 above)
Comment (brandenads) [reply BattleSim/src/main/java/com/revature/models/Roles.java:20]: Verified that the two above constructors are added. Approved.

**Review by plboronski — CHANGES_REQUESTED** [review]

File: BattleSim/src/main/java/com/revature/repositories/RolesRepository.java (hunk 3 above)
Comment (brandenads) [reply BattleSim/src/main/java/com/revature/repositories/RolesRepository.java]: It looks like this plural from updateRoles and deleteRoles is still here. Suggest updateRole and deleteRole.

**Review by brandenads — APPROVED** [review]
Good job.

**Pull request merged**